modules/nf-core/nanocomp/main.nf

```diff
@@ -0,0 +1,129 @@
+process NANOCOMP {
+    label 'process_medium'
+
+    conda "bioconda:nanocomp=1.21.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/nanocomp:1.21.0--pyhdfd78af_0':
+        'quay.io/biocontainers/nanocomp:1.21.0--pyhdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(filelist)
+
+    output:
+    tuple val(meta), path("*NanoComp-report.html"), emit: nanocomp_report_html
+    tuple val(meta), path("*NanoComp_lengths_violin.html"), emit: nanocomp_lengths_violin_html
+    tuple val(meta), path("*NanoComp_log_length_violin.html"), emit: nanocomp_log_length_violin_html
+    tuple val(meta), path("*NanoComp_N50.html"), emit: nanocomp_n50_html
+    tuple val(meta), path("*NanoComp_number_of_reads.html"), emit: nanocomp_number_of_reads_html
+    tuple val(meta), path("*NanoComp_OverlayHistogram.html"), emit: nanocomp_overlay_histogram_html
+    tuple val(meta), path("*NanoComp_OverlayHistogram_Normalized.html"), emit: nanocomp_overlay_histogram_normalized_html
+    tuple val(meta), path("*NanoComp_OverlayLogHistogram.html"), emit: nanocomp_overlay_log_histogram_html
+    tuple val(meta), path("*NanoComp_OverlayLogHistogram_Normalized.html"), emit: nanocomp_overlay_log_histogram_normalized_html
+    tuple val(meta), path("*NanoComp_total_throughput.html"), emit: nanocomp_total_throughput_html
+    tuple val(meta), path("*NanoComp_quals_violin.html"), emit: nanocomp_quals_violin_html, optional: true
+    tuple val(meta), path("*NanoComp_OverlayHistogram_Identity.html"), emit: nanocomp_overlay_histogram_identity_html, optional: true
+    tuple val(meta), path("*NanoComp_OverlayHistogram_PhredScore.html"), emit: nanocomp_overlay_histogram_phredscore_html, optional: true
+    tuple val(meta), path("*NanoComp_percentIdentity_violin.html"), emit: nanocomp_percent_identity_violin_html, optional: true
+    tuple val(meta), path("*NanoComp_ActivePoresOverTime.html"), emit: nanocomp_active_pores_over_time_html, optional: true
+    tuple val(meta), path("*NanoComp_CumulativeYieldPlot_Gigabases.html"), emit: nanocomp_cumulative_yield_plot_gigabases_html, optional: true
+    tuple val(meta), path("*NanoComp_sequencing_speed_over_time.html"), emit: nanocomp_sequencing_speed_over_time_html, optional: true
+    tuple val(meta), path("*NanoStats.txt"), emit: nanocomp_stats_txt
```

Comment:
Can you change the `NanoComp` to value specified in `prefix`? This avoids file name collisions later on :)

Reply:
Also `nanocomp` doesn't need to be part of the output channel name. It's long enough as it is :)

Reply:
> Can you change the `NanoComp` to value specified in `prefix`? This avoids file name collisions later on :)
The "NanoComp" in the file name is used every time, but you can specify a --prefix VALUE in NanoComp command, I did not use it because the user can specify it in args since it is not necessary for the execution. The user could add it, and the output would support that since there is an * in front of every file. Would that be enough or should I add something nonetheless? And if i should, the "prefix" value is stored where? :)

Reply:
Aaah can you add the `--prefix ${prefix}` flag? The reason `prefix` exists is exactly for this :)